src/main/java/appeng/fluids/helper/DualityFluidInterface.java

```diff
@@ -0,0 +1,632 @@
+	{
+		if( channel == AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) )
+		{
+			if( this.hasConfig() )
+			{
+				return null;
+			}
+
+			return (IMEMonitor<T>) this.items;
+		}
+		else if( channel == AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) )
+		{
+			if( this.hasConfig() )
+			{
+				return (IMEMonitor<T>) new InterfaceInventory( this );
+			}
+
+			return (IMEMonitor<T>) this.fluids;
+		}
+
+		return null;
+	}
+
+	public IStorageMonitorable getMonitorable( final IActionSource src )
+	{
+		if( Platform.canAccess( this.gridProxy, src ) )
+		{
+			return this;
+		}
+
+		return null;
+	}
+
+	@Override
+	public TickingRequest getTickingRequest( final IGridNode node )
+	{
+		return new TickingRequest( TickRates.Interface.getMin(), TickRates.Interface.getMax(), !this.hasWorkToDo(), true );
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( final IGridNode node, final int ticksSinceLastCall )
+	{
+		if( !this.gridProxy.isActive() )
+		{
+			return TickRateModulation.SLEEP;
+		}
+
+		final boolean couldDoWork = this.updateStorage();
+		return this.hasWorkToDo() ? ( couldDoWork ? TickRateModulation.URGENT : TickRateModulation.SLOWER ) : TickRateModulation.SLEEP;
+	}
+
+	public void notifyNeighbors()
+	{
+		if( this.gridProxy.isActive() )
+		{
+			try
+			{
+				this.gridProxy.getTick().wakeDevice( this.gridProxy.getNode() );
+			}
+			catch( final GridAccessException e )
+			{
+				// :P
+			}
+		}
+
+		final TileEntity te = this.iHost.getTileEntity();
+		if( te != null && te.getWorld() != null )
+		{
+			Platform.notifyBlocksOfNeighbors( te.getWorld(), te.getPos() );
+		}
+	}
+
+	public void markDirty()
+	{
+		// rescan inventory
+		for( IFluidHandler ih : this.tanks )
+		{
+			onFluidInventoryChanged( ih );
+		}
+	}
+
+	public void gridChanged()
+	{
+		try
+		{
+			this.items.setInternal( this.gridProxy.getStorage().getInventory( AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) ) );
+			this.fluids.setInternal( this.gridProxy.getStorage().getInventory( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) ) );
+		}
+		catch( final GridAccessException gae )
+		{
+			this.items.setInternal( new NullInventory<IAEItemStack>() );
+			this.fluids.setInternal( new NullInventory<IAEFluidStack>() );
+		}
+
+		this.notifyNeighbors();
+	}
+
+	public AECableType getCableConnectionType( final AEPartLocation dir )
+	{
+		return AECableType.SMART;
+	}
+
+	public DimensionalCoord getLocation()
+	{
+		return new DimensionalCoord( this.iHost.getTileEntity() );
+	}
+
+	public boolean hasCapability( Capability<?> capabilityClass, EnumFacing facing )
+	{
+		return capabilityClass == CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY || capabilityClass == Capabilities.STORAGE_MONITORABLE_ACCESSOR;
+	}
+
+	@SuppressWarnings( "unchecked" )
+	public <T> T getCapability( Capability<T> capabilityClass, EnumFacing facing )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "facing". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)